### DISCUSSION TITLE
建议使用 prettier-eslint-cli 格式化代码

### DISCUSSION THREAD

**Discussion by timongh** [the1812/Bilibili-Evolved#3729] [功能建议 / Ideas]
vscode 的自带格式化感觉有点弱。单用 prettier 又无法与当前使用的 eslint 规则完全无冲突。所以建议用 prettier-eslint-cli，它的原理大概是先使用 prettier，再用 eslint --fix。其实我个人倒是挺想弄个 npm 命令把整个项目给格式化了的……

**Comment by timongh** [comment]
对了，如果要使用这个的话最好把 eslint 的 max-len 规则关掉，只在 prettier 的配置中限制长度。因为 prettier [不会严格遵守](https://github.com/prettier/prettier/issues/2564#issuecomment-318916698)最大行长的规则。

**Comment by the1812** [comment]
我是想把 prettier 规则直接配进 ESLint 里的, 但好像对 Vue 2 的 SFC 支持不太好...

**Reply by timongh** [reply]
这个我确实不了解了

**Reply by the1812** [reply]
反正既然你又提了, 我就再研究下, 能用一个工具搞定最好 (至少在 React 项目中我是成功这么弄了的)

**Reply by timongh** [reply]
辛苦辛苦

**Reply by the1812** [reply]
好像还真成了, 我先推到 [eslint-prettier](https://github.com/the1812/Bilibili-Evolved/tree/eslint-prettier) 分支了.
- 不知道为啥按目录调用 ESLint 的时候, `/* eslint-disable */` 不起作用, 不过用到这个的只有几个编译生成的 `.js` 文件, 我就直接取消对 `.js` 的检查了.
- 顺便把 yarn 换成 pnpm 了.

**Reply by timongh** [reply]
挺好，我平时用的也是 pnpm。

另外，tasks 里可以加再点任务：`"pnpm prettier --write ${relativeFile}"`、`"pnpm prettier --write ./**/*.{js|ts|vue|md|css|scss|json|json5|yaml|yml}"`。或者把 prettier 设置成 vscode 的默认格式化方式。顺便一提 prettier 也有一个 `.prettierignore` 文件可以用。至于 CI 里用不用，你随意吧。

**Reply by the1812** [reply]
调用还是用 ESLint，只是我把 prettier 的规则加进去了

**Reply by timongh** [reply]
行
